File: LeetCode/mergeStrings.js
// https://leetcode.com/problems/merge-strings-alternately/?envType=study-plan-v2&envId=leetcode-75

/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
const mergeAlternately = function(word1, word2) {
    let m = word1.split('').length
    let n = word2.split('').length
    let i = 0
    let j = 0
    let results = ''

    while(i < m || j < n) {
        if (i < m) {
            results += (word1[i])
            i++
        }
        if (j < n) {
            results += (word2[j])
            j++
        }
    }
    return results
    
};

let word1 = "abc";
let word2 = "pqr"

console.log(mergeAlternately(word1, word2)) // apbqcr

File: LeetCode/75/Array/merge_strings.py
# https://leetcode.com/problems/merge-strings-alternately/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        m, n = len(word1), len(word2)
        r, l = 0, 0
        result = ''
        while(r < m or l < n):
            if r < m:
                result += word1[r]
                r += 1
            if l < n:
                result += word2[l]
                l += 1
        return result 

            



test = Solution()
print(test.mergeAlternately('abc', 'pqr'))